test/functional/p2p_orphan_handling.py

```diff
@@ -0,0 +1,439 @@
+        child = self.wallet.create_self_transfer(utxo_to_spend=parent['new_utxo'])
+        return child["tx"].getwtxid(), child["tx"], parent["tx"]
+
+    def fastforward(self, seconds):
+        """Convenience helper function to fast-forward, so we don't need to keep track of the
+        starting time when we call setmocktime."""
+        self.mocktime += seconds
+        self.nodes[0].setmocktime(self.mocktime)
+
+    def relay_transaction(self, peer, tx):
+        """Relay transaction using MSG_WTX"""
+        wtxid = int(tx.getwtxid(), 16)
+        peer.send_and_ping(msg_inv([CInv(t=MSG_WTX, h=wtxid)]))
+        self.fastforward(TXREQUEST_TIME_SKIP)
+        peer.wait_for_getdata([wtxid])
+        peer.send_and_ping(msg_tx(tx))
+
+    @cleanup
+    def test_arrival_timing_orphan(self):
+        self.log.info("Test orphan handling delays")
+        node = self.nodes[0]
+        tx_real = self.wallet.create_self_transfer()
+        # Fake orphan spends a nonexistent output of tx_real
+        tx_fake_orphan = self.wallet.create_self_transfer(
+            utxo_to_spend={"txid": tx_real["txid"], "vout": 10, "value": tx_real["new_utxo"]["value"]}
+        )
+        # Real orphan with its real parent
+        orphan_wtxid, orphan_tx, parent_tx = self.create_parent_and_child()
+
+        peer_spy = node.add_p2p_connection(PeerTxRelayer())
+        # This transaction is an orphan because it is missing inputs.
+        # The node should not immediately respond with a request for orphan parents.
+        # Also, no request should be sent later because it will be resolved by
+        # the time the request is scheduled to be sent.
+        peer_spy.assert_message_ignored(msg_tx(tx_fake_orphan["tx"]))
+        # This transaction is also an orphan. Its parent should be requested later.
+        peer_spy.assert_message_ignored(msg_tx(orphan_tx))
+
+        # Node receives transaction. It attempts to obfuscate the exact timing at which this
+        # transaction entered its mempool.
+        node.sendrawtransaction(tx_real["hex"])
+        # Spy peer should not be able to query the node for the parent yet, since it hasn't been
+        # announced / insufficient time has elapsed.
+        parent_inv = CInv(t=MSG_WTX, h=int(tx_real["tx"].getwtxid(), 16))
+        peer_spy.assert_message_ignored(msg_getdata([parent_inv]))
+
+        # The real orphan's parent should be requested, but not until the request delay elapses.
+        peer_spy.assert_never_requested(int(parent_tx.rehash(), 16))
+        self.fastforward(NONPREF_PEER_TX_DELAY)
+        peer_spy.assert_never_requested(int(parent_tx.rehash(), 16))
+        # Request is scheduled with this delay because it is by txid and this
+        # not a preferred relay peer.
+        self.fastforward(TXID_RELAY_DELAY)
+        peer_spy.sync_with_ping()
+
+        self.log.info("Test parent requests don't reveal whether the parent was present when the orphan arrived")
```

Comment:
I don't get this case. Why wouldn't you request a "fake" parent if you don't know it's fake already?